File: c/data_structures/tree/binaryTree.c
#include <stdio.h>
#include <stdlib.h>

struct node {
    struct node *left;
    struct node *right;
    int value;
};

struct node *makeNode(int value) {
    struct node *n;
    n = malloc(sizeof(struct node));
    n->value = value;

    return n;
}

File: javascript/data_structures/tree/binaryTree.js
'use strict';

const makeNode = (l, r, val) => {
    let left = l;
    let right = r;
    let value = val;

    const node = {};

    Object.defineProperties(node, {
        left: {
            enumerable: false,
            get: () => left,
            set: v => left = v
        },
        right: {
            enumerable: false,
            get: () => right,
            set: v => right = v
        },
        value: {
            enumerable: false,
            get: () => value,
            set: v => value = v
        }
    });

    return node;
};

module.exports = makeNode;

